manifests/proxydhcp.pp

```diff
@@ -32,13 +32,18 @@
   }
 
   class { 'dhcp':
-    dnsdomain   => $foreman_proxy::dhcp_option_domain,
-    nameservers => $nameservers,
-    interfaces  => [$foreman_proxy::dhcp_interface] + $foreman_proxy::dhcp_additional_interfaces,
-    pxeserver   => $ip,
-    pxefilename => $foreman_proxy::dhcp_pxefilename,
-    omapi_name  => $foreman_proxy::dhcp_key_name,
-    omapi_key   => $foreman_proxy::dhcp_key_secret,
+    dnsdomain       => $foreman_proxy::dhcp_option_domain,
+    nameservers     => $nameservers,
+    interfaces      => [$foreman_proxy::dhcp_interface] + $foreman_proxy::dhcp_additional_interfaces,
+    pxeserver       => $ip,
+    pxefilename     => $foreman_proxy::dhcp_pxefilename,
+    omapi_name      => $foreman_proxy::dhcp_key_name,
+    omapi_key       => $foreman_proxy::dhcp_key_secret,
+    conf_dir_mode   => '0750',
+    dhcp_root_group => $foreman_proxy::dhcp_manage_acls ? {
```

Comment:
Why do we need to set the group? With ACLs you don't need to and can leave it out.

Reply:
for systems, that do not use acls. Given we have parameter, shouldn't we keep this for users who opt out the acls?

Reply:
Well, it's complicated.

Historically we introduced the ACLs only for Red Hat-based systems because there RPMs reverted the permissions back to 0750. So unless you reran Puppet after yum update (not a problem in regular Puppet deployments), things would break. With this, we actually want to deviate from the OS defaults.

However, you're right that on FreeBSD we end up not managing the DHCP directory so that would be an invalid combination. On the other hand, we don't want to set `/usr/local/etc` to be owned by `foreman-proxy`.

I think we don't want to set the `dhcp_root_group` at all. On Debian and Red Hat I think we can assume ACLs work. On BSD's, the DHCP dir is (correctly) not managed. For BSDs, managing the file permissions on dhcp.conf is probably the only thing you can reasonably do.

Reply:
Well with the `puppet-dhcp` on `master` we would result in changing the root_group only for `dhcp.conf`, because we unmanaged the `dhcp_dir`, so the change to `dhcp_root_group` will not affect the dir.

On other systems, if you would set `$dhcp_manage_acls = false` you'd get to trouble without setting the `dhcp_root_group`. So if we won't manage the group, we need to force the acls and in such case I believe we should drop it from user options and force it for RHEL/Debian and just leave it out for BSD